### PR TITLE
Remove unused locals in System.ServiceModel.Syndication

### PR DESCRIPTION
**Pull request by nxtn**
Contributes to #30457

Only attributes' names are checked here.

### CODE REVIEW

File: src/libraries/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/Atom10FeedFormatter.cs [hunk 1]
```diff
@@ -595,8 +595,6 @@ private static TextSyndicationContent ReadTextContentFromHelper(XmlReader reader
                 }
             }
             reader.MoveToElement();
-            string localName = reader.LocalName;
-            string nameSpace = reader.NamespaceURI;
```
Comment (stephentoub) [inline src/libraries/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/Atom10FeedFormatter.cs]: Do either LocalName or NamespaceURI have side-effects, e.g. advancing the reader, throwing exceptions if something couldn't be parsed, etc.?

Comment (NextTurn) [reply src/libraries/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/Atom10FeedFormatter.cs]: > advancing the reader

`LocalName`/`NamespaceURI` makes the reader query information of the current **node** so there is no streaming.

> throwing exceptions if something couldn't be parsed

`LocalName` and `NamespaceURI` are documented to not throw XML (parsing) exceptions.

**Comment by stephentoub** [comment]
cc: @StephenMolloy @HongGit

**Comment by StephenBonikowsky** [comment]
Looks good to @StephenMolloy and @HongGit !

**Comment by NextTurn** [comment]
Only 1 area label per issue, thanks!

**Pull request merged**